presto-spark-base/src/main/java/com/facebook/presto/spark/execution/http/PrestoSparkHttpWorkerClient.java

```diff
@@ -128,6 +149,48 @@ public Throwable rewriteException(Throwable throwable)
         return null;
     }
 
+    public ListenableFuture<BaseResponse<TaskInfo>> getTaskInfo()
+    {
+        URI uri = uriBuilderFrom(location)
```

Comment:
we're building the URI objects several times in the same way, should we move it to a class member and shared btw methods?

Reply:
Sounds like a plan